maintenance/troubleshoot/commands.md

```diff
@@ -0,0 +1,499 @@
+
+kubectl exec -it multitool -- bash
+```
+```
+bash-5.0# ping 8.8.8.8
+PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
+64 bytes from 8.8.8.8: icmp_seq=1 ttl=97 time=6.61 ms
+64 bytes from 8.8.8.8: icmp_seq=2 ttl=97 time=6.64 ms
+```
+
+#### Collect {{site.prodname}} diagnostic logs
+
+```bash
+sudo calicoctl node diags
+```
+
+```
+Collecting diagnostics
+Using temp dir: /tmp/calico194224816
+Dumping netstat
+Dumping routes (IPv4)
+Dumping routes (IPv6)
+Dumping interface info (IPv4)
+Dumping interface info (IPv6)
+Dumping iptables (IPv4)
+Dumping iptables (IPv6)
+
+Diags saved to /tmp/calico194224816/diags-20201127_010117.tar.gz
+```
+
+### Kubernetes 
+
+
+#### Verify all pods are running 
+
+```bash
+kubectl get pods -A
+```
+
+```
+kube-system       coredns-66bff467f8-dxbtl                   1/1     Running   0          27h
+kube-system       coredns-66bff467f8-n95vq                   1/1     Running   0          27h
+kube-system       etcd-ip-10-0-0-10                          1/1     Running   0          27h
+kube-system       kube-apiserver-ip-10-0-0-10                1/1     Running   0          27h
+```
+
+#### Verify Kubernetes API server is running
+
+```bash
+kubectl cluster-info
+```
+```
+Kubernetes master is running at https://10.0.0.10:6443
+KubeDNS is running at https://10.0.0.10:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+ubuntu@master:~$ kubectl get svc
+NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+kubernetes   ClusterIP   10.49.0.1    <none>        443/TCP   2d2h
+```
+
+#### Verify Kubernetes kube-dns is working
+
+```bash
+kubectl get svc
+```
+```
+NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+kubernetes   ClusterIP   10.49.0.1    <none>        443/TCP   2d2h
+```
+
+```bash
+kubectl exec -it multitool  bash
+```
+```
+bash-5.0# curl -I -k https://kubernetes
+HTTP/2 403 
+cache-control: no-cache, private
+content-type: application/json
+x-content-type-options: nosniff
+content-length: 234
+```
+
+```bash
+bash-5.0# nslookup google.com
+```
+
+```
+Server:         10.49.0.10
+Address:        10.49.0.10#53
+Non-authoritative answer:
+Name:   google.com
+Address: 172.217.14.238
+Name:   google.com
+Address: 2607:f8b0:400a:804::200e
+```
+
+#### Verify that kubelet is running on the node with the correct flags
+
+```bash
+systemctl status kubelet
+```
+If there is a problem, check the journal
+
+```bash
+journalclt -u kubelet | head
+```
+
+#### Check the status of other system pods
+
+Look especially at coredns; if they are not getting an IP, something is wrong with the CNI
+
+```bash
+kubectl get pod -n kube-system -o wide
+```
+But if other pods fail, it is likely a different issue. Perform normal Kubernetes troubleshooting. For example:
+
+```bash
+kubectl describe pod kube-scheduler-ip-10-0-1-20.eu-west-1.compute.internal -n kube-system | tail -15
+``` 
+
+### Calico components
+
+#### View cluster CNI, IPAM, and policy 
```

Comment:
This command doesn't show policies, and will only work on clusters that have Calico installed.

This should probably be "View Calico CNI configuration on a node"

Reply:
Fixed.